File: 2_POO vs Estruturada_alunos/2_comparacao-php_alunos/varios_poo.php
<?php
// ==========================================
// PARTE 2: PROGRAMAÇÃO ORIENTADA A OBJETOS
// ==========================================


// 1ª Digitação (Aqui)


// Criando vários cachorros
// $cachorros = [
//     new Cachorro("Cléo", 247, true),
//     new Cachorro("Francisco", 5, true),
//     new Cachorro("Conan", 19, false),
//     new Cachorro("Leleco", 3, false),
//     new Cachorro("Babi", 13, true)
// ];

// Usando métodos para manipular os objetos


// 2ª Digitação (Aqui)


// Exibindo os resultados no navegador
echo "<!DOCTYPE html>
<html lang='pt-BR'>
<head>
    <meta charset='UTF-8'>
    <meta name='viewport' content='width=device-width, initial-scale=1.0'>
    <title>Resultados dos Cachorros (POO)</title>
    <style>
        body {
            font-family: Arial, sans-serif;
            margin: 20px;
        }
        h1 {
            color: #333;
        }
        p {
            margin: 10px 0;
        }
    </style>
</head>
<body>
    <h1>Resultados dos Cachorros (POO)</h1>";

foreach ($cachorros as $cachorro) {
    echo "<p><strong>{$cachorro->getNome()}</strong>:</p>";
    echo "<ul>";
    echo "<li>Comida restante: <strong>{$cachorro->getComida()}</strong></li>";
    echo "<li>Está com sono? <strong>" . ($cachorro->getSono() ? 'Sim' : 'Não') . "</strong></li>";
    echo "</ul>";
}

echo "</body>
</html>";
?>

File: 2_POO vs Estruturada_alunos/Python/varios_poo.py
class Cachorro:
    def __init__(self, nome, comida, sono):
        self.nome = nome
        self.comida = comida
        self.sono = sono

    def comer(self):
        self.comida = max(self.comida - 1, 0)

    def dormir(self):
        self.sono = True

    def exibir_status(self):
        print(f"{self.nome} agora tem {self.comida} unidades de comida.")
        print(f"{self.nome} está com sono? {'Sim' if self.sono else 'Não'}")

# Criando vários cachorros
cachorros = [
    Cachorro("Nelson", 3, False),
    Cachorro("Jeremias", 1, True),
    Cachorro("Cléo", 247, True),
    Cachorro("Francisco", 5, True),
    Cachorro("Conan", 19, False),
    Cachorro("Leleco", 3, False),
    Cachorro("Babi", 13, True)
]

# Manipulando os cachorros
for cachorro in cachorros:
    cachorro.comer()

# Exibindo os resultados no terminal
for cachorro in cachorros:
    cachorro.exibir_status()